src/quantum/index-32bit.js

```diff
@@ -108,326 +109,39 @@ export default class QuantumIndex32 extends React.Component {
-          [
-            <PerfherderGraphContainer
-              title='Tab Switch (tps)'
-              series={[
-                {
-                  label: 'Firefox',
-                  seriesConfig: {
-                    extraOptions: ['e10s', 'stylo'],
-                    frameworkId: 1,
-                    platform: 'windows7-32',
-                    option: 'pgo',
-                    project: 'mozilla-central',
-                    suite: 'tps',
-                  },
-                },
-              ]}
-            />,
-            <PerfherderGraphContainer
-              title='SVG (tsvg_static)'
-              series={[
-                {
-                  label: 'Firefox',
-                  seriesConfig: {
-                    extraOptions: ['e10s', 'stylo'],
-                    frameworkId: 1,
-                    platform: 'windows7-32',
-                    option: 'pgo',
-                    project: 'mozilla-central',
-                    suite: 'tsvg_static',
-                  },
-                },
-              ]}
-            />,
-          ],
-          [
-            <PerfherderGraphContainer
-              title='SVG (tsvgr_opacity)'
-              series={[
-                {
-                  label: 'Firefox',
-                  seriesConfig: {
-                    extraOptions: ['e10s', 'stylo'],
-                    frameworkId: 1,
-                    platform: 'windows7-32',
-                    option: 'pgo',
-                    project: 'mozilla-central',
-                    suite: 'tsvgr_opacity',
-                  },
-                },
-              ]}
-            />,
-            <PerfherderGraphContainer
-              title='SVG (tsvgx)'
-              series={[
-                {
-                  label: 'Firefox',
-                  seriesConfig: {
-                    extraOptions: ['e10s', 'stylo'],
-                    frameworkId: 1,
-                    platform: 'windows7-32',
-                    option: 'pgo',
-                    project: 'mozilla-central',
-                    suite: 'tsvgx',
-                  },
-                },
-              ]}
-            />,
-          ],
-        ],
+        title: '#3 Pages',
+        rows:
+          _
+          .chain(PAGES.data)
+          // ZIP HEADER WITH ROWS TO GET OBJECTS
+          .map(row => _.zipObject(PAGES.header, row))
+          // GROUP BY title
+          .groupBy(row => row.title)
+          // LOOP OVER EACH KEY/VALUE
+          .toPairs()
+          .map(([title, series], i) => {
+            return (
+              <PerfherderGraphContainer
+                key={`pages${i}`}
+                title={title}
+                series={
+                _
+                .chain(series)
+                  .map((s) => {
+                    return {
+                      label: s.label,
+                      seriesConfig: s,
+                    };
+                  })
+                  .value()
+              }
```

Comment:
If you do this in the config:
```javascript
const CONFIG = {
   'Tp6: YouTube': {
      title='Tp6: YouTube',
      series=[
                {
                  label: 'Firefox',
                  seriesConfig: {
                    frameworkId: 10,
                    platform: 'windows7-32',
                    option: 'pgo',
                    project: 'mozilla-central',
                    suite: 'raptor-tp6-youtube-firefox',
                  },
                },
                {
                  label: 'Chrome',
                  seriesConfig: {
                    frameworkId: 10,
                    platform: 'windows7-32-nightly',
                    option: 'opt',
                    project: 'mozilla-central',
                    suite: 'raptor-tp6-youtube-chrome',
                  },
        ],
}
```
you can do the following here:
```javascript
Object.entries(config).map((key, value) => (
  <PerfherderGraphContainer
     key={key}
     ...value
   />
)
```

If you do not want to have 2 configs you can then remove `platform` from the configs and insert it at this point like this:
```javascript
Object.entries(config).map((key, value) => {
  const series = Object.assign(value, {});
  // TODO: Iterate through series and add the platform property
  // if the suite contains 'chrome' then also append '-nightly'
  return (<PerfherderGraphContainer
     key={key}
     title={value.title}
     series={series}
   />);
}
```

There's no need to have two graphs per row. We can just set a fix width and rely on flexing for this.
You can use this guide for help:
https://css-tricks.com/snippets/css/a-guide-to-flexbox/

Reply:
Putting the structured call pattern required for `<PerfherderGraphContainer>` into `CONFIG` only makes things more complicated: The information is not anonymous, it is not inlined, it has a name; which is another layer of indirection.

The `CONFIG` is an attempt to separate data from layout and style.  It has a few benefits over the  deep structured form used by `<PerfherderGraphContainer>`:

1. This form is centralized, and reduces the risk that the 32bit and 64bit pages deviate in some way.  It is step towards merging those two pages.
2. The `CONFIG` data is flatter, so it fits in a table, and it can be formatted to reveal its redundancy, *or important exceptions*. The deep structured form does not reveal these patterns or exceptions unless there is meticulous accounting of all the charts; even then something may be missed: did one of the charts used a special config?  This accounting is difficult using the `<PerfherderGraphContainer>` format because it spans hundreds of lines, and is mostly boilerplate with little signal. 
3. With `CONFIG` being flat, the patterns and redundancy is easy to spot: `CONFIG` shows off the fact that all the Chrome tests are run on one config, and Firefox is all run on another, except for the test in `frameworkId==1`, which appear to run on `['e10s', 'stylo']`.  We can take further steps to remove functional data dependencies to make it simpler.  This redundancy is not removed in this pull request, but it is a start to reveal there are patterns in the data. 
4. The `CONFIG` object is data, and the choice of data (the choice of tests shown) may have a pattern:  If we know this pattern then we can write some code that pulls all the tests based on a rule, rather than manually maintaining this `CONFIG`.  When we pull this `CONFIG` data dynamically, then the code need only provide simple, flat, data independent of the `<PerfherderGraphContainer>` calling conventions. 
5. With the data separate from the layout, we can see `<PerfherderGraphContainer>` is just one link in a chain of conversion functions:
    * `perf-goggles.queryPerformanceData()` transforms data, which gets passed to
    * `getPerfherderData()` which transforms the data and gets passed to
    * `perfherderFormatter()` which transforms the data and passes it to  
    * `<PerfherderGraphContainer>` which annotates the data with 
    * `generateOptions()` and then gets pass to 
    * `<ChartJsWrapper>` which transforms the data, and passes it to 
    * `<Chart>` which, ostensibly, does the drawing

    This chain, was hard to follow.  Each named function alludes to the fact some other code, not found in this chain, may need its transformation. But that is not the case: this is an exclusive chain.  At the very least this chain should be made from anonymous functions, or limited-scoped named, functions, or merged into just one function, so the API surface area is minimized.  
6. By separating `CONFIG` as data, we can start pointing to the major features of any  BI/dashboarding tool; which helps with code navigation:  Business rules, showing what tests are important, and what platforms those test run on, are in CONFIG.  The layout and style is  demonstrated by the `index32bit.js`, and pages in general.  A Chart API uses a standard data format that is consumed by `<ChartJsWrapper>`, or any other charting lib.  Storage accessed by  `queryPerformanceData()` (a Query API) which should be returning something close to perfherder's native format.
![image](https://user-images.githubusercontent.com/2334429/49096388-792eca00-f238-11e8-8dc8-c212b515e8c2.png)

The `src/quantum/config.js` looked like a good spot to start adding the quantum business rules.